galaxyproject/galaxy
# [23.0] Filter category

Fix for issue #12244

Changed function browse_repositories_in_category to pass in the category name as a filter, and also added it as a filtering column in RepositoriesInCategoryGrid

~~Also, replaced a usage of `favicon.png` with `favicon.ico` in the top bar, as this caused a visual bug. The live toolshed at https://toolshed.g2.bx.psu.edu/ appears to retrieve `favicon.ico`, so I feel this was a regression somewhere down the line.~~
^ Reverted in favor of #15246 

## How to test the changes?
  1. Create multiple categories, with one having more repositories than the page limit.
  2. Open toolshed, and click "Browse by category"
  3. Click on the desired category, and browse through the pages, asserting that it remains within that category. 
  4. Go back, and inspect other categories

The network request being made when clicking on page 2 was:
```
localhost:9009/repository/browse_repositories_in_category?async=false&sort=name&page=2&show_item_checkboxes=false&advanced_search=false&f-Category.name=Metabolomics
```

Screenshots: https://imgur.com/a/jKfTwd8
Shows 2 categories, and page 1 and 2 of `Metabolomics`

Unfortunately, I couldn't get a fully functional unit test working since the way toolshed handles pagination is rather roundabout, and selenium tests don't appear to be set up in this part of the code. Since the code being changed is specific to this endpoint, I do not thing this would cause any regressions in other parts of the code.

## License
